cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -156,6 +156,27 @@ func (g *gitHubRepository) ComponentsPath() string {
 
 // GetFile returns a file for a given provider version.
 func (g *gitHubRepository) GetFile(ctx context.Context, version, path string) ([]byte, error) {
+	log := logf.Log
+
+	cacheID := fmt.Sprintf("%s/%s:%s:%s", g.owner, g.repository, version, path)
+	if content, ok := cacheFiles[cacheID]; ok {
+		return content, nil
+	}
+
+	// Try to get the file using http get.
+	// NOTE: this can be disabled by setting GORPOXY to `direct` or `off` (same knobs used for skipping goproxy requests).
+	if goProxyClient, _ := g.getGoproxyClient(ctx); goProxyClient != nil {
```

Comment:
We're getting this client just to check GOPROXY, but then don't use it?

Why is GOPROXY relevant for this code?

Reply:
I just used the same knobs we are already using to disable other optimization on the GitHub provider.

Reply:
Alright, let's go with that